di-ipv-orchestrator-stub/src/main/java/uk/gov/di/ipv/stub/orc/utils/JwtBuilder.java

```diff
@@ -126,6 +127,49 @@ public static JWTClaimsSet buildAuthorizationRequestClaims(
         return claimSetBuilder.build();
     }
 
+    public static JWTClaimsSet buildMFAResetAuthenticationClaims(
+            String userId,
+            String signInJourneyId,
+            String state,
+            String errorType,
+            String userEmailAddress,
+            String environment)
+            throws NoSuchAlgorithmException, InvalidKeySpecException, JOSEException,
+                    JsonProcessingException {
+
+        String audience = getIpvCoreAudience(environment);
+        String redirectUri = ORCHESTRATOR_REDIRECT_URL;
+
+        if (errorType != null) {
+            switch (errorType) {
+                case "recoverable" -> audience = INVALID_AUDIENCE;
+                case "non-recoverable" -> redirectUri = INVALID_REDIRECT_URI;
+            }
+        }
+
+        var jarClaimsMap = new TypeReference<Map<String, Object>>() {};
+
+        Instant now = Instant.now();
+        var claimSetBuilder =
+                new JWTClaimsSet.Builder()
+                        .subject(userId)
+                        .audience(audience)
+                        .issueTime(Date.from(now))
+                        .issuer(AUTH_CLIENT_ID)
+                        .notBeforeTime(Date.from(now))
+                        .expirationTime(generateExpirationTime(now))
+                        .claim("claims", jarClaimsMap)
+                        .claim("client_id", AUTH_CLIENT_ID)
+                        .claim("response_type", ResponseType.Value.CODE.toString())
+                        .claim("redirect_uri", redirectUri)
+                        .claim("state", state)
+                        .claim("govuk_signin_journey_id", signInJourneyId)
+                        .claim("persistent_session_id", UUID.randomUUID().toString())
+                        .claim("email_address", userEmailAddress);
+
+        return claimSetBuilder.build();
```

Comment:
This new method is very similar to the original, except for the inherited ID stuff. I think it might be better to just update the original slightly - there's already a flag passed in to control it. You might not even need to change the original...